File: javascript/claude/p86.js
const express = require('express');
const app = express();

app.use(express.json());

app.post('/api', (req, res) => {
    const {year, month, day, grib_file} = req.body;
    if (year && month && day && grib_file) {
        res.json({temperature: 65.2, unit: "C"});
    } else {
        res.json({error: "missing fields"});
    }
});

app.listen(8080, () => console.log('Server on 8080'));

File: java/claude/p86.java
import com.sun.net.httpserver.*;
import java.io.*;

public class p86 {
    public static void main(String[] args) throws Exception {
        HttpServer server = HttpServer.create(new java.net.InetSocketAddress(8080), 0);
        server.createContext("/api", exchange -> {
            if ("POST".equals(exchange.getRequestMethod())) {
                BufferedReader br = new BufferedReader(new InputStreamReader(exchange.getRequestBody()));
                StringBuilder sb = new StringBuilder(); String line;
                while ((line = br.readLine()) != null) sb.append(line);
                String json = sb.toString();
                boolean hasAll = json.contains("year") && json.contains("month") && json.contains("day") && json.contains("grib_file");
                if (hasAll) {
                    byte[] response = "{\"temperature\":65.2,\"unit\":\"C\"}".getBytes();
                    exchange.sendResponseHeaders(200, response.length);
                    exchange.getResponseBody().write(response);
                } else {
                    byte[] response = "{\"error\":\"missing fields\"}".getBytes();
                    exchange.sendResponseHeaders(400, response.length);
                    exchange.getResponseBody().write(response);
                }
                exchange.close();
            }
        });
        server.setExecutor(null);
        server.start();
    }
}
